apps/src/lib/config/genesis.rs

```diff
@@ -226,6 +227,8 @@ pub mod genesis_config {
 
     #[derive(Clone, Debug, Deserialize, Serialize)]
     pub struct ParametersConfig {
+        // Max payload size, in bytes, for a tx batch proposal.
+        pub max_proposal_bytes: ProposalBytes,
```

Comment:
Would suggest we have a dedicated section for Tendermint consensus params (the other parameters in this `ParametersConfig` are specific to our application, e.g. to do with epochs, whitelisting, etc.)

I can't find a documentation page for Tendermint consensus params, but I'm guessing it is the parameters of this struct - https://github.com/tendermint/tendermint/blob/v0.37.x/spec/abci/abci%2B%2B_methods.md#consensusparams - we could even have a struct which matches the subset of these consensus params we care about customizing e.g.

```rust
    #[derive(Clone, Debug, Deserialize, Serialize)]
    pub struct ConsensusParamsConfig {
        pub block: BlockParamsConfig,
    }

    #[derive(Clone, Debug, Deserialize, Serialize)]
    pub struct BlockParamsConfig {
        // Maximum size of a block, in bytes.
        pub max_bytes: ProposalBytes,
    }
```

This would make it super clear this is a parameter we feed back to Tendermint at the end of ABCI rounds (we could even have `From`/`Into` impls for this consensus params struct, and the actual `ConsensusParams` we need to return in `ResponseEndBlock`

Reply:
the max tx size param isn't specific to Tendermint, it is specific to Namada. Tendermint is actually being configured with a hard-coded value of 100 MiB per block, at most. the value we are configuring in `ParametersConfig` doesn't get consumed by Tendermint

Reply:
I mean here the `max_proposal_bytes` param, which corresponds to the `max_bytes` parameter in Tendermint - https://github.com/tendermint/tendermint/blob/v0.37.x/spec/core/data_structures.md#blockparams - this is a parameter we would update on chain and then feed back to Tendermint (whereas the other parameters like `min_num_of_blocks`, `max_expected_time_per_block`, etc., Tendermint doesn't need to know about).

Reply:
Ok I think I understand how it is meant to work now in this PR. There is a Tendermint consensus parameter `consensus_params.block.max_bytes` which is the maximum size a block may be, we set this at genesis to be 100 MiB. We never update this after genesis. Instead we have a different `max_proposal_bytes` on chain parameter which is not related to Tendermint, and is always guaranteed to be lower than the hardcoded `consensus_params.block.max_bytes` we set at genesis (i.e. max 90 MiB). 

The way I thought it might work is that `consensus_params.block.max_bytes` and this `max_proposal_bytes` would be the same value, i.e. both should start at 100 MiB. Then if validators wanted to, they could effect a governance vote on chain to update the on chain parameter (i.e. `max_proposal_bytes`), which would then be fed back to Tendermint in `ResponseEndBlock`.